.github/workflows/main.yml

```diff
@@ -6,8 +6,20 @@ on:
     branches: [master]
 
 jobs:
+  formatting-check:
+    name: Formatting Check
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Run clang-format style check on src directory
+        uses: jidicula/clang-format-action@v4.3.0
+        with:
+          clang-format-version: "13"
+          check-path: "src"
+
   build:
     runs-on: ubuntu-20.04
+    needs: [formatting-check]
```

Comment:
why? They are 2 independent things. Also if we want to move formatting-check to main.yml, then you need to remove that file in the same commit

Reply:
I think the idea is :
Formating is mandatory so if formating check failed, the pr will need update so that isn't worse it to do full check for something that will need correction anyway.

Reply:
CI will still fail and PR will need to be corrected.
I believe it's important to know if code builds and runs regardless of any coding format issue (that can be easily fixed up later)

Reply:
Besides @khancyr comment, software style is usually mandatory to be fixed, the check runs in 20 seconds and limit the build resource (the github actions machines) to be free and build others PRs.